src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/OurShould.java

```diff
@@ -24,37 +29,99 @@ public OurShould(OurWebElement element, OurSearchStrategy strategy) {
-            try {
-                return !element.isDisplayed();
-            } catch (Throwable ignored) {
-                //in case of any exception in OurWebElement considering that element is absent
-                return true;
+        waitFor(new ExpectedCondition<Boolean>() {
+            @Override
+            public Boolean apply(WebDriver input) {
+                try {
+                    return !element.isDisplayed();
+                } catch (Throwable ignored) {
+                    //in case of any exception in OurWebElement considering that element is absent
+                    return true;
+                }
+            }
+
+            @Override
+            public String toString() {
+                return String.format("should element '%s' be absent", element.toString());
             }
         });
     }
 
     public void haveText(String text) {
-        waitFor((element) -> element.getText().equals(text));
+        waitFor(new ExpectedCondition<Boolean>() {
+            @Override
+            public Boolean apply(WebDriver input) {
+                return element.getText().equals(text);
+            }
+
+            @Override
+            public String toString() {
+                return String.format("should element '%s' have text '%s'", element.toString(), text);
+            }
+        });
     }
 
     public void haveAttribute(String attributeName, String value) {
-        waitFor((element) -> element.getAttribute(attributeName).equals(value));
+        waitFor(new ExpectedCondition<Boolean>() {
+            @Override
+            public Boolean apply(WebDriver input) {
+                return element.getAttribute(attributeName).equals(value);
+            }
+
+            @Override
+            public String toString() {
+                return String.format("should element '%s' have attribute '%s' with value '%s'", element.toString(), attributeName, value);
+            }
+        });
+    }
+
+    public void haveAttribute(String attributeName) {
+        waitFor(new ExpectedCondition<Boolean>() {
+            @Override
+            public Boolean apply(WebDriver input) {
+                return element.getAttribute(attributeName) != null;
+            }
+
+            @Override
+            public String toString() {
+                return String.format("should element '%s' have attribute '%s'", element.toString(), attributeName);
+            }
+        });
     }
 
     public void notHaveAttribute(String attributeName) {
-        waitFor((element) -> element.getAttribute(attributeName) == null);
+        waitFor(new ExpectedCondition<Boolean>() {
+            @Override
+            public Boolean apply(WebDriver input) {
+                return element.getAttribute(attributeName) == null;
+            }
+
+            @Override
+            public String toString() {
+                return String.format("should element '%s' not have attribute '%s'", element.toString(), attributeName);
```

Comment:
here and everywhere please replace "should element" to "element should" for better wording